src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderTransactionHistoryResourceTest.java

```diff
@@ -0,0 +1,338 @@
+package uk.gov.justice.hmpps.prison.api.resource.impl;
```

Comment:
I think it's missing a test case for the Offender not found.

Reply:
it's an empty result set for which there is a test

Reply:
Should it be Not Found instead?  I've heard arguments for and against not found vs empty list and so it's debatable, but I think in prison-api historically we've gone for a Not Found.

Reply:
so if no records are found it should return HTTP 404? isn't that simple wrong? because the resource does exist it just doesn't have any records associated with it? just checking before I make the change

Reply:
Sorry, should have been clearer.
Offender not found - 404
Offender found but no transaction history - 200 with empty list

Reply:
so no change required because as it is now if you pass an offender number that doesn't exist then it'll return an empty list and if you pass a non-number value then you get an error, agree?

Reply:
> I think it's missing a test case for the Offender not found.

> it's an empty result set for which there is a test

So do we need an a test for Offender not found bearing in mind it's not an empty result set?

Reply:
so how is it not found and also not empty :)

Reply:
that can only happen if the caller uses the wrong id number from the one they intended, which we can't detect, is my point

Reply:
There is no test for a 404 response.

Reply:
sure, my question is how would you detect or test for that?

Reply:
See offender damage obligations resource test, I recently added a test for an offender not found and an offender not in the current user's caseload, both return a 404.

Reply:
done